circuits/src/types/wallet.rs

```diff
@@ -55,30 +70,34 @@ where
     pub fees: [Fee; MAX_FEES],
     /// The key tuple used by the wallet; i.e. (pk_root, pk_match, pk_settle, pk_view)
     pub keys: PublicKeyChain,
-    /// The wallet randomness used to blind commitments, nullifiers, etc
+    /// The wallet randomness used to blind secret shares
     #[serde(
         serialize_with = "scalar_to_hex_string",
         deserialize_with = "scalar_from_hex_string"
     )]
-    pub randomness: Scalar,
+    pub blinder: Scalar,
 }
 
 /// Represents a wallet that has been allocated in a constraint system
 #[derive(Clone, Debug)]
-pub struct WalletVar<const MAX_BALANCES: usize, const MAX_ORDERS: usize, const MAX_FEES: usize>
-where
+pub struct WalletVar<
+    const MAX_BALANCES: usize,
+    const MAX_ORDERS: usize,
+    const MAX_FEES: usize,
+    L: Into<LinearCombination>,
+> where
     [(); MAX_BALANCES + MAX_ORDERS + MAX_FEES]: Sized,
 {
     /// The list of balances in the wallet
-    pub balances: [BalanceVar; MAX_BALANCES],
+    pub balances: [BalanceVar<L>; MAX_BALANCES],
     /// The list of open orders in the wallet
-    pub orders: [OrderVar; MAX_ORDERS],
+    pub orders: [OrderVar<L>; MAX_ORDERS],
     /// The list of payable fees in the wallet
-    pub fees: [FeeVar; MAX_FEES],
+    pub fees: [FeeVar<L>; MAX_FEES],
     /// The key tuple used by the wallet; i.e. (pk_root, pk_match, pk_settle, pk_view)
-    pub keys: PublicKeyChainVar,
-    /// The wallet randomness used to blind commitments, nullifiers, etc
-    pub randomness: Variable,
+    pub keys: PublicKeyChainVar<L>,
+    /// The wallet randomness used to blind secret shares
+    pub blinder: Variable,
```

Comment:
Should this be `L` instead of `Variable`?

Reply:
yeah, good catch, fixed